src/main/scala/viper/gobra/frontend/PackageResolver.scala

```diff
@@ -31,6 +31,8 @@ object PackageResolver {
   val stubDirectories = Vector("stubs")
   val fileUriScheme = "file"
   val jarUriScheme = "jar"
+  // packages ending with "_test" are considered test packages
+  private val testPackageRegex = """(.+)_test$""".r
```

Comment:
The documentation is very weak here. But in practice, the go compiler enforces that you can only compile together files belonging to package X and X_test. I checked this with the following minimal example:

file f.go:
```go
package p
```

file f_test.go:
```go
package p_test
```

Compiling these two files together with the command `go build f.go g_test.go` compiles the package. If instead you rewrite file f_test.go to the following, you get an error:
```go
package p_bla
```

Error:
```
found packages p (f.go) and p_bla (f_test.go) in /Users/joao/Desktop/test/test/test_test
```

This does not rule out necessarily that test files can have arbitrary names, but it does at least indicate that packages ending with `_test` have some special treatment. Unfortunately, I did not find unambiguous official sources that claim this.

Reply:
> When we are already at it, we could also add a special case for a directory named testdata that should completely be removed (search for "testdata" [here](https://pkg.go.dev/cmd/go@master#hdr-Test_packages)). In particular, this would help me as I'm currently facing a Gobra error because of not ignoring files in such directories

What kinds of errors are you having? Wouldn't gobra ignore these folders, as they usually do not have any .go and .gobra files?

Reply:
> What kinds of errors are you having? Wouldn't gobra ignore these folders, as they usually do not have any .go and .gobra files?

I get the following error in package `golang.org/x/tools`:
`Missing package clause in ./.modules/golang.org/x/tools/cmd/fiximports/testdata/src/old.com/bad/bad.go`
I think it's a folder of invalid go files (for testing purposes) and thus have a `.go` file ending ^^

Reply:
> This does not rule out necessarily that test files can have arbitrary names, but it does at least indicate that packages ending with `_test` have some special treatment. Unfortunately, I did not find unambiguous official sources that claim this.

Thanks for this additional bit of information. I would thus rephrase my observation that Go treats `testdata` folder and files ending in `_test.go` differently and additionally enforces that packages in files ending in `_test.go` cannot have arbitrary package clauses by must either be the same one as in the other files in this folder or ending in `_test`.

I'm not sure though to what extend we want to handle this in Gobra.  It seems however more desirable to watch out for `_test.go` files and completely ignore them (we would sacrifice the error if you would use an invalid package clause in there but technically that does only matter for compilation but not really at runtime).
Only checking the package clause for a `_test` ending seems less desirable to me as we could end up treating a file as being part of the codebase even though it's a unit test used for whitebox testing (and thus has the same package clause as the actual source files)

Reply:
Haha but unfortunately not because it's throwing a violation before filtering the packages (as it could not identify the package clause as there is none)